### ISSUE TITLE
async completion: eats existing code on commit

### ISSUE THREAD

**Issue by DavidKarlas** [dotnet/roslyn#33626]
**Version Used**: 
VS 2019 Preview 3
**Steps to Reproduce**:

1.  Create standard Console app
2. type `override` above `static void Main..` method
3. create space and press Ctrl+Space to trigger code completion
4. type `asdasdasdasdasdasd` or any random text
5. select one of code completion options and press Enter

**Expected Behavior**:
Nicely inserted new code
**Actual Behavior**:
<img width="439" alt="image" src="https://user-images.githubusercontent.com/774791/53282110-7d815b00-3733-11e9-80ea-1002d7b7b96f.png">

**Comment by KirillOsenkov** [comment]
@jinujoseph @ivanbasov @AmadeusW

**Comment by DavidKarlas** [comment]
Just wanted to clarify one thing, this repro steps work on VS Windows and VS Mac.

**Comment by dpoeschl** [comment]
This is bad. Min repro + some notes:

1. Paste C#:
```C#
class C
{
    override $$
    void M() { }
}
```
2. Remove the $$ and at that location, trigger completion, and type `GetHashCode`
3. Commit completion (e.g. with <kbd>Tab</kbd>)

**Result**: The override is inserted, but the subsequent `"GetHashCode".Length` characters are deleted. So in this case, 11 characters get deleted -- 2 for the newline, and then 9 more -- leaving this:
```C#
class C
{
    public override int GetHashCode()
    {
        return base.GetHashCode();
    }M() { }
}
```

@ivanbasov I'm confused about this: 
https://github.com/dotnet/roslyn/blob/ec366687c8caead948e58704ad61ed644eb91b5b/src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs#L173

Basically, the `triggerSnapshot` there contains just `override `, while the `textChange` was produced a couple lines earlier using the _current_ `document` which includes `override GetHashCode`, so things are mismatched and it falls apart.

However, if the above is the problem, then I'm surprised that it isn't also impacting the other `CompletionProvider`s with custom `GetChangeAsync` methods, including the doc comment and IVT providers. I tried them both and couldn't repro the problem.

**Comment by jinujoseph** [comment]
@dpoeschl are you testing this in preview4 ?
Am i missing a repro here 

![completion](https://user-images.githubusercontent.com/12449387/53283267-87d23380-36f8-11e9-91ab-27c01c2bdd53.gif)

![completion1](https://user-images.githubusercontent.com/12449387/53283268-8e60ab00-36f8-11e9-9f7b-281a1c1b67e6.gif)

**Comment by DavidKarlas** [comment]
@jinujoseph try typing much longer string

**Comment by jinujoseph** [comment]
@DavidKarlas , @dpoeschl  yes , was able to repro when i try to type longer string  like  GetHashcode and commit

**Comment by ivanbasov** [comment]
Hmm....I'm on 16.0 preview 4 (28622.103). Also tried a previous preview 4 version. Both work fine for me:
![override](https://user-images.githubusercontent.com/5455484/53365172-6ee2a180-38f5-11e9-8eca-568fb28711b4.gif)

**Comment by sharwell** [comment]
@ivanbasov This bug is being caught by integration tests

**Comment by ivanbasov** [comment]
I found how to reproduce it: need to type move letters before Tab. Looking into it.

**Comment by jkotalik** [comment]
I filed an issue on VS for this: https://developercommunity.visualstudio.com/content/problem/466995/auto-complete-is-incorrectly-formatted-for-method.html. Check the internal ticket for it. The repro steps shows a few issues (which I think both are rooted in the same issue).

- If I try completing a method, the method arguments are placed on the next line rather than the same line.
- If I misspell the method (capitalization, for example), auto complete creates an invalid method.

**Comment by jinujoseph** [comment]
addressed via #33666
